artichoke-backend/src/extn/core/string/trampoline.rs

```diff
@@ -410,6 +418,74 @@ pub fn byteslice(
     length: Option<Value>,
 ) -> Result<Value, Error> {
     let s = unsafe { super::String::unbox_from_value(&mut value, interp)? };
+    let maybe_range = if length.is_none() {
+        index.is_range(interp, s.char_len() as i64)?
```

Comment:
```suggestion
        index.is_range(interp, s.len() as i64)?
```

since we're doing byte slices.